reader_concurrency_semaphore.cc

```diff
@@ -150,7 +150,8 @@ class reader_permit::impl
 
 private:
     reader_concurrency_semaphore& _semaphore;
-    const schema* _schema;
+    const schema_ptr _schema;
```

Comment:
We can store non-const `schema_ptr` here. This is what we do everywhere in the code. The `schema` object is immutable by design anyway.
Also, in the constructors and methods, we can drop the `const`, it just interferes with `std::move()`.